.github/workflows/pre_publish.yml

```
@@ -80,12 +80,36 @@ jobs:
       # TODO @Thoralf-M  get version, bump it, update it
       # TODO @Thoralf-M generate changelog
 
-      # - name: Create PR for Python
-      #   if: inputs.package == 'python-sdk'
-      #   run: |
-      #     echo "Creating PR for Python SDK - Not yet implemented."
-      # TODO @alex  get version, bump it, update it
-      # TODO @alex generate changelog
+      - name: Bump Python version
+        if: inputs.package == 'iota-python-sdk'
+        id: bump
+        run: |
+          # Read current version from pyproject.toml
+          CONFIG="bindings/python/src/pyproject.toml"
+          CURRENT_VERSION_PEP440=$(grep '^version = ' $CONFIG | sed 's/version = "\(.*\)"/\1/')
+
+          source ./.github/bump_version.sh
+          CURRENT_VERSION=$(pep440_to_semver "$CURRENT_VERSION_PEP440")
+          echo "Current version: $CURRENT_VERSION"
+
+          # Bump the version using the input level
+          NEW_VERSION=$(bump_semver "$CURRENT_VERSION" "${{ inputs.level || 'alpha' }}")
```

```suggestion
          NEW_VERSION=$(bump_semver "$CURRENT_VERSION" "${{ inputs.level }}")
```
did you want to keep this for a specific reason? In my branch I only added it for testing

---

no, I thought it's necessary and missed some information.